### ISSUE TITLE
List Collection

### ISSUE THREAD

**Issue by jweinst1** [jweinst1/Wind#12]
## Intro

`Wind` should have a form of ordered collection called a `List`. The list in wind should be able to hold values in an ordered sequence, one that is indexable, supports random access, expansion, and slicing in reasonable amounts of time. 

### Example

One example is creating a type called `WindElement` that serves as a member of the list, and then using an iterative begin and end style struct

```c
struct WindElement
{
        WindType type;
        WindValue value;
};

typedef struct WindElement WindElement;

struct WindList
{
        WindElement* begin;
        WindElement* end;
};

typedef struct WindList WindList;
```

**Comment by jweinst1** [comment]
Another issue is a mutable sequence vs immutable sequence, and in the context of a stream language, what is really needed.

For an alternative option, `Wind` could separate collection types into expandable vs fixed sized, and mutable vs immutable.

**Comment by jweinst1** [comment]
Implemented in #13
